src/Core/Framework/Api/OAuth/Client/ApiClient.php

```diff
@@ -17,6 +17,7 @@ class ApiClient implements ClientEntityInterface
     public function __construct(
         private readonly string $identifier,
         private readonly bool $writeAccess,
+        private readonly bool $confidential,
```

Comment:
Btw: isn't this a breaking change as neither the constructor nor the class itself is declared `internal`? 😬🙈

Reply:
I was under the impression that constructors are not part of the public API

Reply:
https://developer.shopware.com/docs/resources/guidelines/code/public-apis.html I guess this does fall under DTO's , so you are right. We could introduce a setter instead maybe: `setConfidential()` - wdyt @mitelg @keulinho or anyone else ? :)

Reply:
:thinking: we can make it nullable, and then required for the next major version. if it passed as `null` we throw/trigger a deprecation. 

but then we need to set a non breaking default, which is `false` if I understood it correctly?

Reply:
The previous value was always `true` and that seems like a more reasonable default to have for `isConfidential` 🤔

Reply:
Puh i guess this is only public by accident, all the implementations are super tightly coupled to oauth lib anyways 

Let's not break it when we can keep compatible with a default value

We should consider marking most of the oauth stuff as internal 🤔 
But we don't need to do that as part of this PR

Reply:
@MalteJanz yes sorry default was true, I've updated. With the new version though, the default does not work for all grant types. See the discussion in the linked PR. As I understand it, it's not about safety.